[CI] StandardVersusLogsIndexModeChallengeRestIT testTermsAggregation failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3580 / almalinux-8-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/iltt2r7ux6czk)
- [elasticsearch-pull-request #27297 / part-1](https://gradle-enterprise.elastic.co/s/ew3lhpwbwmoty)
- [elasticsearch-periodic-platform-support #3576 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/ie3vacvvks3zw)
- [elasticsearch-periodic-platform-support #3576 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/gvkg4sy64l5c4)

**Reproduction Line:**
```
./gradlew ":modules:data-streams:javaRestTest" --tests "org.elasticsearch.datastreams.logsdb.qa.StandardVersusLogsIndexModeChallengeRestIT.testTermsAggregation" -Dtests.seed=D7DCE5621A863818 -Dtests.locale=mk -Dtests.timezone=Africa/Douala -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testTermsAggregation'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.datastreams.logsdb.qa.StandardVersusLogsIndexModeChallengeRestIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Error [Lists do not match when ignoring sort order] actual mappings [{"_doc":{"properties":{"host.name":{"type":"keyword"},"gvuYK":{"type":"keyword"},"L":{"type":"long"},"OCGAZrjc":{"type":"keyword"},"PBjCTj":{"type":"keyword"}}}}] actual settings [{"index":{"mode":"logsdb","number_of_shards":"3","mapping":{"total_fields":{"limit":"5000"}}}}] expected mappings [{"_doc":{"subobjects":false,"properties":{"host.name":{"type":"keyword"},"gvuYK":{"type":"keyword"},"L":{"type":"long"},"OCGAZrjc":{"type":"keyword"},"PBjCTj":{"type":"keyword"}}}}] expected settings [{"index":{"number_of_replicas":"2","mapping":{"total_fields":{"limit":"5000"}}}}] 
```

**Issue Reasons:**
- [main] 4 failures in test testTermsAggregation (2.1% fail rate in 189 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/AbstractChallengeRestTest.java
+++ b/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/AbstractChallengeRestTest.java
@@ -186,18 +186,22 @@ public abstract class AbstractChallengeRestTest extends ESRestTestCase {
         assert deleteContenderTemplate.getStatusLine().getStatusCode() == RestStatus.OK.getStatus();
     }
 
-    private Settings.Builder createSettings(final CheckedConsumer<Settings.Builder, IOException> settingsConsumer) throws IOException {
+    private Settings.Builder createSettings(
+        final CheckedConsumer<Settings.Builder, IOException> settingsConsumer,
+        final CheckedConsumer<Settings.Builder, IOException> commonSettingsConsumer
+    ) throws IOException {
         final Settings.Builder settings = Settings.builder();
         settingsConsumer.accept(settings);
+        commonSettingsConsumer.accept(settings);
         return settings;
     }
 
     private Settings.Builder createBaselineSettings() throws IOException {
-        return createSettings(this::baselineSettings);
+        return createSettings(this::baselineSettings, this::commonSettings);
     }
 
     private Settings.Builder createContenderSettings() throws IOException {
-        return createSettings(this::contenderSettings);
+        return createSettings(this::contenderSettings, this::commonSettings);
     }
 
     private XContentBuilder createMappings(final CheckedConsumer<XContentBuilder, IOException> builderConsumer) throws IOException {
@@ -221,6 +225,8 @@ public abstract class AbstractChallengeRestTest extends ESRestTestCase {
     public void baselineSettings(Settings.Builder builder) {}
 
     public void contenderSettings(Settings.Builder builder) {}
+
+    public void commonSettings(Settings.Builder builder) {}
 
     private Response indexDocuments(
         final String dataStreamName,

--- a/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusLogsIndexModeChallengeRestIT.java
+++ b/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusLogsIndexModeChallengeRestIT.java
@@ -49,6 +49,8 @@ import static org.hamcrest.Matchers.greaterThan;
 
 public class StandardVersusLogsIndexModeChallengeRestIT extends AbstractChallengeRestTest {
     private final DataGenerator dataGenerator;
+    private final int numShards = randomBoolean() ? randomIntBetween(2, 5) : 0;
+    private final int numReplicas = randomBoolean() ? randomIntBetween(1, 3) : 0;
 
     public StandardVersusLogsIndexModeChallengeRestIT() {
         super("standard-apache-baseline", "logs-apache-contender", "baseline-template", "contender-template", 101, 101);
@@ -112,27 +114,24 @@ public class StandardVersusLogsIndexModeChallengeRestIT extends AbstractChalleng
         }
     }
 
-    private static void settings(final Settings.Builder settings) {
-        if (randomBoolean()) {
-            settings.put("index.number_of_shards", randomIntBetween(2, 5));
+    @Override
+    public void commonSettings(Settings.Builder builder) {
+        if (numShards > 0) {
+            builder.put("index.number_of_shards", numShards);
         }
-        if (randomBoolean()) {
-            settings.put("index.number_of_replicas", randomIntBetween(1, 3));
+        if (numReplicas > 0) {
+            builder.put("index.number_of_replicas", numReplicas);
         }
+        builder.put("index.mapping.total_fields.limit", 5000);
     }
 
     @Override
     public void contenderSettings(Settings.Builder builder) {
         builder.put("index.mode", "logsdb");
-        builder.put("index.mapping.total_fields.limit", 5000);
-        settings(builder);
     }
 
     @Override
-    public void baselineSettings(Settings.Builder builder) {
-        builder.put("index.mapping.total_fields.limit", 5000);
-        settings(builder);
-    }
+    public void baselineSettings(Settings.Builder builder) {}
 
     @Override
     public void beforeStart() throws Exception {
@@ -223,7 +222,7 @@ public class StandardVersusLogsIndexModeChallengeRestIT extends AbstractChalleng
 
         final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery())
             .size(0)
-            .aggregation(new TermsAggregationBuilder("agg").field("host.name"));
+            .aggregation(new TermsAggregationBuilder("agg").field("host.name").size(numberOfDocuments));
 
         final MatchResult matchResult = Matcher.mappings(getContenderMappings(), getBaselineMappings())
             .settings(getContenderSettings(), getBaselineSettings())

--- a/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/ArrayEqualMatcher.java
+++ b/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/ArrayEqualMatcher.java
@@ -35,7 +35,13 @@ class ArrayEqualMatcher extends EqualMatcher<Object[]> {
     private MatchResult matchArraysEqual(final Object[] actualArray, final Object[] expectedArray, boolean ignoreSorting) {
         if (actualArray.length != expectedArray.length) {
             return MatchResult.noMatch(
-                formatErrorMessage(actualMappings, actualSettings, expectedMappings, expectedSettings, "Array lengths do no match")
+                formatErrorMessage(
+                    actualMappings,
+                    actualSettings,
+                    expectedMappings,
+                    expectedSettings,
+                    "Array lengths do no match, " + prettyPrintArrays(actualArray, expectedArray)
+                )
             );
         }
         if (ignoreSorting) {
@@ -47,14 +53,20 @@ class ArrayEqualMatcher extends EqualMatcher<Object[]> {
                         actualSettings,
                         expectedMappings,
                         expectedSettings,
-                        "Arrays do not match when ignoring sort order"
+                        "Arrays do not match when ignoring sort order, " + prettyPrintArrays(actualArray, expectedArray)
                     )
                 );
         } else {
             return matchArraysEqualExact(actualArray, expectedArray)
                 ? MatchResult.match()
                 : MatchResult.noMatch(
-                    formatErrorMessage(actualMappings, actualSettings, expectedMappings, expectedSettings, "Arrays do not match exactly")
+                    formatErrorMessage(
+                        actualMappings,
+                        actualSettings,
+                        expectedMappings,
+                        expectedSettings,
+                        "Arrays do not match exactly, " + prettyPrintArrays(actualArray, expectedArray)
+                    )
                 );
         }
     }

--- a/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/ListEqualMatcher.java
+++ b/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/ListEqualMatcher.java
@@ -35,7 +35,13 @@ class ListEqualMatcher extends EqualMatcher<List<?>> {
     private MatchResult matchListEquals(final List<Object> actualList, final List<Object> expectedList, boolean ignoreSorting) {
         if (actualList.size() != expectedList.size()) {
             return MatchResult.noMatch(
-                formatErrorMessage(actualMappings, actualSettings, expectedMappings, expectedSettings, "List lengths do no match")
+                formatErrorMessage(
+                    actualMappings,
+                    actualSettings,
+                    expectedMappings,
+                    expectedSettings,
+                    "List lengths do no match, " + prettyPrintLists(actualList, expectedList)
+                )
             );
         }
         if (ignoreSorting) {
@@ -47,14 +53,20 @@ class ListEqualMatcher extends EqualMatcher<List<?>> {
                         actualSettings,
                         expectedMappings,
                         expectedSettings,
-                        "Lists do not match when ignoring sort order"
+                        "Lists do not match when ignoring sort order, " + prettyPrintLists(actualList, expectedList)
                     )
                 );
         } else {
             return matchListsEqualExact(actualList, expectedList)
                 ? MatchResult.match()
                 : MatchResult.noMatch(
-                    formatErrorMessage(actualMappings, actualSettings, expectedMappings, expectedSettings, "Lists do not match exactly")
+                    formatErrorMessage(
+                        actualMappings,
+                        actualSettings,
+                        expectedMappings,
+                        expectedSettings,
+                        "Lists do not match exactly, " + prettyPrintLists(actualList, expectedList)
+                    )
                 );
         }
     }

--- a/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/Matcher.java
+++ b/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/Matcher.java
@@ -12,6 +12,10 @@ import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.xcontent.XContentBuilder;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
 /**
  * A base class to be used for the matching logic when comparing query results.
  */
@@ -104,4 +108,15 @@ public abstract class Matcher {
             + "] ";
     }
 
+    protected static String prettyPrintArrays(final Object[] actualArray, final Object[] expectedArray) {
+        return "actual: " + prettyPrintList(Arrays.asList(actualArray)) + ", expected: " + prettyPrintList(Arrays.asList(expectedArray));
+    }
+
+    protected static String prettyPrintLists(final List<Object> actualList, final List<Object> expectedList) {
+        return "actual: " + prettyPrintList(actualList) + ", expected: " + prettyPrintList(expectedList);
+    }
+
+    private static String prettyPrintList(final List<Object> list) {
+        return "[" + list.stream().map(Object::toString).collect(Collectors.joining(", ")) + "]";
+    }
 }

--- a/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/ObjectMatcher.java
+++ b/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/matchers/ObjectMatcher.java
@@ -28,7 +28,13 @@ public class ObjectMatcher extends EqualMatcher<Object> {
         return actual.equals(expected)
             ? MatchResult.match()
             : MatchResult.noMatch(
-                formatErrorMessage(actualMappings, actualSettings, expectedMappings, expectedSettings, "Actual does not equal expected")
+                formatErrorMessage(
+                    actualMappings,
+                    actualSettings,
+                    expectedMappings,
+                    expectedSettings,
+                    "Actual does not equal expected, actual: " + actual + ", expected: " + expected
+                )
             );
     }
 }

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -156,9 +156,6 @@ tests:
 - class: org.elasticsearch.xpack.restart.FullClusterRestartIT
   method: testDataStreams {cluster=UPGRADED}
   issue: https://github.com/elastic/elasticsearch/issues/111448
-- class: org.elasticsearch.datastreams.logsdb.qa.StandardVersusLogsIndexModeChallengeRestIT
-  method: testTermsAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/111441
 
 # Examples:
 #

